tools/apiview/token-schema/model.tsp

```diff
@@ -0,0 +1,107 @@
+import "@typespec/json-schema";
+
+using TypeSpec.JsonSchema;
+
+@jsonSchema
+
+/** CodeFile represents entire API review object. This will be processed to render review lines. */
+model CodeFile {
+  packageName: string;
+  packageVersion: string;
+  /** version of the APIview language parser used to create token file*/
+  parserVersion: string;
+  language: "C"|"C++"|"C#"|"Go"|"Java"|"JavaScript"|"Kotlin"|"Python"|"Swagger"|"Swift"|"TypeSpec";
+  /** Language variant is applicable only for java variants*/
+  languageVariant?: "None" | "Spring" | "Android" = "None";
+  crossLanguagePackageId?: string;
+  codeLines: Array<CodeLine>;
+  /** Add any system generated comments for the current token. */
+  systemComments?: Array<CodeDiagnostic>;
+}
+
+
+/** Code line object corresponds to each line displayed on API review. If an empty line is required then add a code line object without any token. */
+model CodeLine {
+  /** lineId is only required if we need to support commenting on a line that contains this token. 
+   *  Usually code line for documentation or just punctuation is not required to have lineId. lineId should be a unique value within 
+   *  the review token file to use it assign to review comments as well as navigation Id within the review page.
+   *  for e.g Azure.Core.HttpHeader.Common, azure.template.template_main
+   */
+  lineId?: string;
+  crossLanguageId?: string;
+  /** list of tokens that constructs a line in API review */
+  tokens: Array<Token>;
+  /** Add any child lines as children. For e.g. all classes and namespace level methods are added as a children of namespace(module) level code line. 
+   *  Similarly all method level code lines are added as children of it's class code line.*/
+  children?: Array<CodeLine>;
+}
+
+
+/** Token corresponds to each component within a code line. A separate token is required for keyword, punctuation, type name, text etc. */
+model Token {  
+  kind: TokenKind;
+  value: string;
+  /** Subkind is required when kind is TypeName or Membername */
+  SubKind?: TokenSubKind;
+  /** navigateToId should be set if the underlying token is required to be displayed as HREF to another type within the review.
+   * For e.g. a param type which is class name in the same package
+   */
+  navigateToId?: string;
+  /** set skipDiff to true if underlying token needs to be ignored from diff calculation. For e.g. package metadata or dependency versions 
+   *  are usually excluded when comparing two revisions to avoid reporting them as API changes*/
+  skipDiff?: boolean = false;
+  /** This is set if API is marked as hidden */
+  isHidden?: boolean = false;
```

Comment:
Why is "isHidden" on the token rather than the line? I ask because I'm thinking about how to recreate the TypeSpec navigation. I would have some top-level CodeLines for Resources, Models, etc. but these should not appear in the actual text of the APIView. They _will_ have children. So it seems to me that the CodeLine should be hidden (or not), and if the line is hidden, it will be important to factor that into the automatic indentation. 

Are there use cases for hidden tokens within a line the way there was for hiding things from the diff?

Reply:
I don't see any special use case. I can move it to line model.